packages/issm/package.py

```diff
@@ -1,57 +1,137 @@
-        args.append("--with-mumps-dir={0}".format(self.spec["mumps"].prefix))
-        args.append("--with-m1qn3-dir={0}".format(self.spec["m1qn3"].prefix.lib))
 
-        # Even though we set the MPI compilers manually, the build system
-        # wants us to explicitly request an MPI-enabled build by telling
-        # it the MPI include directory.
+        #
+        # Wrappers
+        #
+        if "+wrappers" in self.spec:
+            args.append("--with-wrappers=yes")
+        else:
+            args.append("--with-wrappers=no")
+
+        #
+        # MPI: Even if we use Spack's MPI wrappers, the build system may 
+        # want explicit mention of MPI includes and compilers:
+        #
         args.append("--with-mpi-include={0}".format(self.spec["mpi"].prefix.include))
         args.append("CC=" + self.spec["mpi"].mpicc)
         args.append("CXX=" + self.spec["mpi"].mpicxx)
         args.append("FC=" + self.spec["mpi"].mpifc)
         args.append("F77=" + self.spec["mpi"].mpif77)
 
+        #
+        # PETSc, MUMPS, METIS, SCALAPACK, etc.
+        # (Spack typically puts these in the compiler/link environment,
+        #  but if the ISSM configure script looks for explicit --with-* 
+        #  flags, we pass them.)
+        #
+        args.append("--with-petsc-dir={0}".format(self.spec["petsc"].prefix))
+        args.append("--with-metis-dir={0}".format(self.spec["metis"].prefix))
+        args.append("--with-mumps-dir={0}".format(self.spec["mumps"].prefix))
+        # If you rely on sca/lapack from PETSc, these lines might 
+        # not be strictly necessary. If ISSM's configure script 
+        # checks them individually, add them:
+        args.append("--with-scalapack-dir={0}".format(self.spec["scalapack"].prefix))
+        args.append("--with-parmetis-dir={0}".format(self.spec["parmetis"].prefix))
+        args.append("--with-triangle-dir={0}".format(self.spec["access-triangle"].prefix))
+
+        #
+        # M1QN3
+        #
+        # Some codes want the actual library subdir for m1qn3, 
+        # e.g. "prefix.lib" or "prefix" depending on the configure logic:
+        #
+        args.append("--with-m1qn3-dir={0}".format(self.spec["m1qn3"].prefix.lib))
+        args.append("--with-python-version=3.9")
+        args.append("--with-python-dir=/apps/python3/3.9.2")
+
+
+        numpy_site_packages = "/apps/python3/3.9.2/lib/python3.9/site-packages/numpy-1.20.0-py3.9-linux-x86_64.egg/numpy"
+        numpy_core_dir = "/apps/python3/3.9.2/lib/python3.9/site-packages/numpy-1.20.0-py3.9-linux-x86_64.egg/numpy"
+        args.append("--with-python-numpy-dir={0}".format(numpy_core_dir))
+        
         return args
+
+    def install(self, spec, prefix):
+        # Run the normal Autotools install logic
+        super().install(spec, prefix)
+
+        # Copy the entire source tree into an additional directory
+        install_tree(self.stage.source_path, prefix.src)
```

Comment:
Are you sure that the ISSM Spack package should install all the source? That's very unusual.